src/marks/text.js

```diff
@@ -39,44 +39,68 @@ export class Text extends Mark {
       defaults
     );
     this.rotate = crotate;
-    this.textAnchor = string(textAnchor);
+    this.textAnchor = impliedString(textAnchor, "middle");
+    this.lineAnchor = keyword(lineAnchor, "lineAnchor", ["top", "middle", "bottom"]);
     this.fontFamily = string(fontFamily);
     this.fontSize = cfontSize;
     this.fontStyle = string(fontStyle);
     this.fontVariant = string(fontVariant);
     this.fontWeight = string(fontWeight);
-    this.dx = string(dx);
-    this.dy = string(dy);
   }
   render(index, {x, y}, channels, dimensions) {
     const {x: X, y: Y, rotate: R, text: T, fontSize: FS} = channels;
     const {width, height, marginTop, marginRight, marginBottom, marginLeft} = dimensions;
-    const {rotate} = this;
+    const {dx, dy, rotate} = this;
     const cx = (marginLeft + width - marginRight) / 2;
     const cy = (marginTop + height - marginBottom) / 2;
     return create("svg:g")
         .call(applyIndirectTextStyles, this, T)
-        .call(applyTransform, x, y, offset, offset)
+        .call(applyTransform, x, y, offset + dx, offset + dy)
         .call(g => g.selectAll()
           .data(index)
           .join("text")
-            .call(applyDirectTextStyles, this)
-            .call(R ? text => text.attr("transform", X && Y ? i => `translate(${X[i]},${Y[i]}) rotate(${R[i]})`
+            .call(applyDirectStyles, this)
+            .attr("transform", R ? (X && Y ? i => `translate(${X[i]},${Y[i]}) rotate(${R[i]})`
                 : X ? i => `translate(${X[i]},${cy}) rotate(${R[i]})`
                 : Y ? i => `translate(${cx},${Y[i]}) rotate(${R[i]})`
                 : i => `translate(${cx},${cy}) rotate(${R[i]})`)
-              : rotate ? text => text.attr("transform", X && Y ? i => `translate(${X[i]},${Y[i]}) rotate(${rotate})`
+              : rotate ? (X && Y ? i => `translate(${X[i]},${Y[i]}) rotate(${rotate})`
                 : X ? i => `translate(${X[i]},${cy}) rotate(${rotate})`
                 : Y ? i => `translate(${cx},${Y[i]}) rotate(${rotate})`
                 : `translate(${cx},${cy}) rotate(${rotate})`)
-              : text => text.attr("x", X ? i => X[i] : cx).attr("y", Y ? i => Y[i] : cy))
+              : (X && Y ? i => `translate(${X[i]},${Y[i]})`
+                : X ? i => `translate(${X[i]},${cy})`
+                : Y ? i => `translate(${cx},${Y[i]})`
+                : `translate(${cx},${cy})`))
             .call(applyAttr, "font-size", FS && (i => FS[i]))
-            .call(applyText, T)
-            .call(applyChannelStyles, this, channels))
+            .call(applyChannelStyles, this, channels)
+            .call(applyMultilineText, this, T))
       .node();
   }
 }
 
+function applyMultilineText(selection, {lineAnchor}, T) {
+  if (!T) return;
+  const format = isTemporal(T) ? isoFormat : isNumeric(T) ? formatNumber() : string;
```

Comment:
We could use `String` here rather than `string` because we know that `T[i]` can never be null or undefined due to filtering.